engine/src/test/java/com/redhat/cloud/notifications/routers/EndpointTestResourceTest.java

```diff
@@ -106,4 +117,82 @@ void testEndpoint() {
+
+        // Only one test action should have been sent to Kafka.
+        final int expectedActionsCount = 1;
+        Assertions.assertEquals(expectedActionsCount, actionsList.size(), "unexpected number of actions sent to Kafka");
+
+        final String kafkaActionRaw = actionsList.get(0).getPayload();
+        final Action kafkaAction = Parser.decode(kafkaActionRaw);
+
+        // Check that the top level values coincide.
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_BUNDLE, kafkaAction.getBundle(), "unexpected bundle in the test action");
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_APPLICATION, kafkaAction.getApplication(), "unexpected application in the test action");
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_EVENT_TYPE, kafkaAction.getEventType(), "unexpected event type in the test action");
+        Assertions.assertEquals(orgId, kafkaAction.getOrgId(), "unexpected org id in the test action");
+
+        final Context context = kafkaAction.getContext();
+        final Map<String, Object> contextProperties = context.getAdditionalProperties();
+        Assertions.assertEquals(createdEndpoint.getId().toString(), contextProperties.get(TestEventHelper.TEST_ACTION_CONTEXT_ENDPOINT_ID), "unexpected endpoint ID received in the action's context");
+
+        // Check the events, their metadata and their payload.
+        final List<Event> events = kafkaAction.getEvents();
+
+        final int expectedEventsCount = 1;
+        Assertions.assertEquals(expectedEventsCount, events.size(), "unexpected number of test action events");
+
+        final Event event = events.get(0);
+
+        final Metadata metadata = event.getMetadata();
+        final Map<String, Object> metadataAdditionalProperties = metadata.getAdditionalProperties();
+
+        final int expectedMetadataAdditionalPropertiesCount = 1;
+        Assertions.assertEquals(expectedMetadataAdditionalPropertiesCount, metadataAdditionalProperties.size(), "unexpected number of metadata additional properties");
+
+        final String metadataValue = (String) metadata.getAdditionalProperties().get(TestEventHelper.TEST_ACTION_METADATA_KEY);
+
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_METADATA_VALUE, metadataValue, "unexpected event metadata value");
+
+        final Payload payload = event.getPayload();
+        final Map<String, Object> payloadAdditionalProperties = payload.getAdditionalProperties();
+
+        final int expectedPayloadAdditionalPropertiesCount = 1;
+        Assertions.assertEquals(expectedPayloadAdditionalPropertiesCount, payloadAdditionalProperties.size(), "unexpected number of payload additional properties");
+
+        final String payloadValue = (String) payload.getAdditionalProperties().get(TestEventHelper.TEST_ACTION_PAYLOAD_KEY);
+
+        Assertions.assertEquals(customMessage, payloadValue, "unexpected event payload value");
+    }
```

Comment:
For some odd reason this test times out waiting for messages. I am guessing I might need to use a different memory sink for this test?

Reply:
FYI I'm still reviewing the tests. I may answer this question soon.